src/main/kotlin/com/wandera/affectedmodulesdetector/task/FindAffectedModulesTask.kt

```diff
@@ -0,0 +1,41 @@
+package com.wandera.affectedmodulesdetector.task
+
+import com.wandera.affectedmodulesdetector.graph.Graph
+import com.wandera.affectedmodulesdetector.graph.Node
+import com.wandera.affectedmodulesdetector.output.FileOutputWriter
+import com.wandera.affectedmodulesdetector.output.LogPrintOutput
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import javax.inject.Inject
+
+/**
+ * Task for finding affected modules.
+ */
+abstract class FindAffectedModulesTask : DefaultTask() {
+
+    @Input
+    lateinit var attrs: FindAffectedModulesAttributes
+
+    @Input
+    lateinit var graph: Graph
+
+    @get:Inject
+    abstract val layout: ProjectLayout
+
+    private lateinit var helper: FindAffectedModulesTaskHelper
+
+    @TaskAction
+    fun findAffectedModules() {
+        helper = FindAffectedModulesTaskHelper(graph)
+
+        LogPrintOutput.printTaskInput(attrs.toString())
+
+        val affectedModules: Set<Node> = helper.getAffectedModules(attrs.baseBranch)
```

Comment:
It seems you don't need a saved variable here at all
```suggestion
        val affectedModules: Set<Node> = FindAffectedModulesTaskHelper(graph).getAffectedModules(attrs.baseBranch)
```